### PR TITLE
Added Accessibility Guide For Terra-Image

### PR DESCRIPTION
**Pull request by supreethmr**
### Summary

Added Accessibility guide for Terra-Image to provide overview of accessibility features that are provided by terra-image.

Closes #

### Deployment Link

https://terra-core-img-accessib-zz4fks.herokuapp.com/components/cerner-terra-core-docs/image/accessibility-guide

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+#### Examples
+<DecorativeImageBar title="Decorative Image"/>
+
+_Note: `DecorativeImage` component sets `alt` prop to empty `""` value internally to make `terra-image` as Decorative Image._
+
+## Usability Expectations:
+
+### Keyboard Navigation
+ 
+Terra Images does not have any keyboard navigation expectations on its own. Any content provided to the `startContent`, `endContent`, or `children` should follow the interaction expectations of those elements.
```
Comment (benbcai) [inline packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx:75]: I see the term "Terra Images" sometimes being used (on line 12) as a plural and here as a singular. Is this referring to the component name (Terra Image) or just images in general?

File: packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+**Decorative images** must have a null (empty) alt text (alt="") so that they can be ignored by assistive technologies, such as screen readers. Text values for these types of images would add audible clutter to screen reader output or could distract users if the topic is different from that in adjacent text. Leaving out the alt attribute is also not an option because when it is not provided, some screen readers will announce the file name of the image instead.
+
+_Whether to treat an image as informative or decorative is a judgment that should be made by Author, based on the reason for including the image on the page._
+
+Following section describes how to use the `alt` attribute of the `Terra-Image` component in various situations.
+
+### 1. Use [Image](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/image).
+- When the `Image` contribute meaning to the current page or context.
```
Comment (benbcai) [inline packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx:42]:
```suggestion
- When the `Image` contributes meaning to the current page or context.
```

File: packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+- When the `Image` is a simple graphic or photograph that describes the intended meaning of image.
+- When the `Image` is conveying an impression or emotion.
```
Comment (benbcai) [inline packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx:44]:
```suggestion
- When the `Image` conveys an impression or emotion.
```

File: packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+**Decorative images** must have a null (empty) alt text (alt="") so that they can be ignored by assistive technologies, such as screen readers. Text values for these types of images would add audible clutter to screen reader output or could distract users if the topic is different from that in adjacent text. Leaving out the alt attribute is also not an option because when it is not provided, some screen readers will announce the file name of the image instead.
+
+_Whether to treat an image as informative or decorative is a judgment that should be made by Author, based on the reason for including the image on the page._
+
+Following section describes how to use the `alt` attribute of the `Terra-Image` component in various situations.
+
+### 1. Use [Image](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/image).
```
Comment (benbcai) [inline packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx:41]:
```suggestion
### 1. Use [Image](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/image)
```

File: packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+See [Informative Images](https://www.w3.org/WAI/tutorials/images/informative/) for more info.
+
+#### Related or Additional WCAG Resources for Informative Image.
+- [G94: Providing short text alternative for non-text content that serves the same purpose and presents the same information as the non-text content](https://www.w3.org/WAI/WCAG21/Techniques/general/G94).
+- [H37: Using alt attributes on img elements](https://www.w3.org/WAI/WCAG21/Techniques/html/H37).
+
+#### Examples
+<InformativeImage />
+
+### 2. Use [DecorativeImage](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/decorative-image).
```
Comment (benbcai) [inline packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx:55]:
```suggestion
### 2. Use [DecorativeImage](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/decorative-image)
```

File: packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+- When `Image` contain text and the text is also present as real text nearby.
+- When `Image` contain text and the text is only shown for visual effects.
+- When the `Image` purely decorative or not intended for the user.
```
Comment (benbcai) [inline packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx:58]:
```suggestion
- When `Image` contains text and the text is also present as real text nearby.
- When `Image` contains text and the text is only shown for visual effects.
- When `Image` is purely decorative or not intended for the user.
```

**Comment by xenoworf** [comment]
@neilpfeiffer  just explained that this PR is quite early in the process, so I'll hold off on further suggestions around grammar/language/formatting. In general this looks fine to me. When it is time to review further I'll be happy to do so. Carry on!

**Comment by neilpfeiffer** [comment]
New changes have been applied and look great.

**Review by neilpfeiffer — APPROVED** [review]
Tested: PR #3592
\
\
**Reviewed:**
- Adds new accessibility guide for Terra Image with sections indicating proper usage for:
  - Why it is important
  - Code Considerations
  - Content Considerations
  - Usability Expectations and Interaction Details
  - Support Compliance
  - Linked References
- Contained examples and explanations for:
  - Informative Images
  - Decorative Images
  
  \
  \
Verified by @neilpfeiffer

 **UX Reviewed** |
-----------|

**Pull request merged**